src/toil/jobStores/azureJobStore.py

```diff
@@ -800,12 +843,23 @@ def attempt(last=False):
         else:
             go.pop()
 
+    total_tries = num_tries
+    totalWaitTime = 0
     while go:
-        if num_tries == 1:
-            yield attempt(last=True)
-        else:
-            yield attempt()
+        try:
+            waitTime = waitTimes[total_tries - num_tries]
+        except IndexError:
+            waitTime = waitTimes[-1]
+        finally:
+            totalWaitTime += waitTime
+
+        yield attempt(last=num_tries == 1,
```

Comment:
Ensure that you don't sleep or "make an attempt" if the sleep would exceed the timeout.